### PR TITLE
Fix VkNonDispatchableHandle alignment on Linux x86

### PR DESCRIPTION
**Pull request by Kangz**
There alignof(uint64_t) is 8 but it is aligned to 4 inside structures.

PTAL @SenorBlanco @kainino0x

### CODE REVIEW

File: src/common/vulkan_platform.h [hunk 1]
```diff
@@ -32,21 +32,41 @@
...
-#    define ALIGNAS_VULKAN_HANDLE alignas(alignof(uint64_t))
+#    define DAWN_DEFINE_NATIVE_NON_DISPATCHABLE_HANDLE(object) using object##Native = uint64_t;
 #else
 #    error "Unsupported platform"
 #endif
 
+// Define a dummy Vulkan handle for use before we include vulkan.h
+DAWN_DEFINE_NATIVE_NON_DISPATCHABLE_HANDLE(VkSomeHandle)
+
+// Find out the alignement of native handles. Logically we would use alignof(VkSomeHandleNative) so
```
Comment (kainino0x) [inline src/common/vulkan_platform.h]: alignement -> alignment

Comment (Kangz) [reply src/common/vulkan_platform.h]: Done

**Comment by Kangz** [comment]
Thanks!

File: src/common/vulkan_platform.h [hunk 2]
```diff
@@ -32,21 +32,41 @@
...
+DAWN_DEFINE_NATIVE_NON_DISPATCHABLE_HANDLE(VkSomeHandle)
+
+// Find out the alignment of native handles. Logically we would use alignof(VkSomeHandleNative) so
+// why bother with the wrapper struct? It turns out that on Linux Intel x86 alignof(uint64_t) is 8
+// but alignof(struct{uint64_t a;}) is 4. This is because this Intel ABI doesn't say anything about
+// double-word alignment so for historical reasons compilers violated the standard and use an
+// alignment of 4 for uint64_t (and double) inside structures.
+// See https://stackoverflow.com/questions/44877185
+// One way to get the alignment inside structures of a type is to look at the alignment of it
+// wrapped in a structure. Hence VkSameHandleNativeWrappe
```
Comment (SenorBlanco) [inline src/common/vulkan_platform.h]: Nit: Wrappe -> Wrapper

**Pull request merged**